project1-sps/script.js

```diff
@@ -1,4 +1,177 @@
+  return messageUsername;
+};
+
+//var userSPSErrorCheck = function (userSPS) {
+// check if input is invalid i.e. not scissors, paper or stones, or the reversed
+// if (
+//   userSPS != "scissors" &&
+//   userSPS != "paper" &&
+//  userSPS != "stone" &&
+//  userSPS != "reversed scissors" &&
+//  userSPS != "reversed paper" &&
+//   userSPS != "reversed stone"
+//  ) {
+//    var messageErrorCheck = `Invalid input. <br> Please select scissors, paper or stone.`;
+// }
+//console.log(`userSPSErrorCheck message ${messageErrorCheck}`);
+//return messageErrorCheck;
+//};
+
+var winOrLoseSPSLogic = function (userSPS, pcSPS) {
+  // draw scenario
+  if (userSPS == pcSPS) {
+    var winOrLose = "It's a draw. ";
+    drawCount += 1;
+  }
+  // you win scenario
+  else if (
+    (userSPS == "scissors" && pcSPS == "paper") ||
+    (userSPS == "paper" && pcSPS == "stone") ||
+    (userSPS == "stone" && pcSPS == "scissors")
+  ) {
+    var winOrLose = "You win! ";
+    userWinCount += 1;
+    winLoseRoundsPlayed += 1;
+  }
+  // you lose scenario
+  else if (
+    (userSPS == "paper" && pcSPS == "scissors") ||
+    (userSPS == "scissors" && pcSPS == "stone") ||
+    (userSPS == "stone" && pcSPS == "paper")
+  ) {
+    var winOrLose = "You Lose! ";
+    pcWinCount += 1;
+    winLoseRoundsPlayed += 1;
+  }
+  return winOrLose;
+};
+
+var winOrLoseReversedSPSLogic = function (userSPS, pcSPS) {
+  // draw scenario
+  if (userSPS == pcSPS) {
+    var winOrLose = "It's a draw. ";
+    drawCount += 1;
+  }
+  // you win scenario
+  else if (
+    (userSPS == "reversed scissors" && pcSPS == "reversed paper") ||
+    (userSPS == "reversed paper" && pcSPS == "reversed stone") ||
+    (userSPS == "reversed stone" && pcSPS == "reversed scissors")
+  ) {
+    var winOrLose = "You lose! ";
+    userWinCount += 1;
+    winLoseRoundsPlayed += 1;
+  }
+  // you lose scenario
+  else if (
+    (userSPS == "reversed paper" && pcSPS == "reversed scissors") ||
+    (userSPS == "reversed scissors" && pcSPS == "reversed stone") ||
+    (userSPS == "reversed stone" && pcSPS == "reversed paper")
+  ) {
+    var winOrLose = "You win! ";
+    pcWinCount += 1;
+    winLoseRoundsPlayed += 1;
+  }
+  return winOrLose;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var userSPS = input;
+  // change username only for first time input
+  message = usernameInputCheck(userSPS);
+
+  // check for error input (for 2nd inputs onwards)
+  if (usernameInputCount > 1) {
+    //Defining game logic
+    var winOrLose = "Nothing happened. ";
+
+    // pc chooses
+    if (
+      userSPS == "reversed scissors" ||
+      userSPS == "reversed paper" ||
+      userSPS == "reversed stone"
+    ) {
+      var pcSPS = chooseSPS();
+      if (pcSPS == 1) {
+        var pcSPS = "reversed scissors";
+      } else if (pcSPS == 2) {
+        var pcSPS = "reversed paper";
+      } else if (pcSPS == 3) {
+        var pcSPS = "reversed stone";
+      }
```

Comment:
In lines 121, 123, 125, we are reassigning the value of pcSPS (you have already defined it in line 119) so no need for the use of keyword var anymore!